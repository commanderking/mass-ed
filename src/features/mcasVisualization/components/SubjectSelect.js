import React from "react";
import { Button, ButtonGroup } from "reactstrap";
import { subjectsConstants } from "../mcasConstants";
export const subjects = [subjectsConstants.ELA, subjectsConstants.MATH];

const isSelectedSubject = (subject, selectedSubject) =>
  subject === selectedSubject;

const formatBackgroundColorForSubject = (subject, selectedSubject) =>
  isSelectedSubject(subject, selectedSubject) ? "#006CD6" : "gray";

const renderSubjectButtons = (setSubject, selectedSubject) =>
  subjects.map(subject => (
    <Button
      key={`subjectSelect-${subject}-button`}
      style={{
        backgroundColor: formatBackgroundColorForSubject(
          subject,
          selectedSubject
        )
      }}
      onClick={() => {
        setSubject(subject);
      }}
    >
      {subject}
    </Button>
  ));

const SubjectSelect = ({ setSubject, selectedSubject }) => (
  <ButtonGroup className="subjectSelect">
    {renderSubjectButtons(setSubject, selectedSubject)}
  </ButtonGroup>
);

export { SubjectSelect };
