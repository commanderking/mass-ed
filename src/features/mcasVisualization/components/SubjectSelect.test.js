import React from "react";
import { shallow } from "enzyme";
import { SubjectSelect, subjects } from "./SubjectSelect";
import { Button, ButtonGroup } from "reactstrap";

describe("SubjectSelect", () => {
  let mockProps = {
    selectedSubject: "ela",
    setSubject: jest.fn()
  };
  let wrapper = shallow(<SubjectSelect {...mockProps} />);

  it("renders correct structure", () => {
    expect(wrapper.find(ButtonGroup)).toHaveLength(1);
    expect(wrapper.find(Button)).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });

  it("fires setSubject action for ela and math", () => {
    const buttonWrapper = wrapper.find(Button);
    buttonWrapper.first().simulate("click");
    expect(mockProps.setSubject).toHaveBeenCalledTimes(1);
    expect(mockProps.setSubject).toHaveBeenCalledWith(subjects[0]);
  });
});
