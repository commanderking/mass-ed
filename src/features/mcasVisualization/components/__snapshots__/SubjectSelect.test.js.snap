// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectSelect renders correct structure 1`] = `
<ButtonGroup
  className="subjectSelect"
  role="group"
  tag="div"
>
  <Button
    color="secondary"
    key="subjectSelect-ELA-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
    tag="button"
  >
    ELA
  </Button>
  <Button
    color="secondary"
    key="subjectSelect-MATH-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
    tag="button"
  >
    MATH
  </Button>
</ButtonGroup>
`;
